staging/src/k8s.io/client-go/examples/fake-client/main_test.go

```diff
@@ -31,7 +31,8 @@ import (
 // TestFakeClient demonstrates how to use a fake client with SharedInformerFactory in tests.
 func TestFakeClient(t *testing.T) {
 	// Use a timeout to keep the test from hanging.
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
wait.ForeverTestTimeout

Reply:
this is not using timeout, just the function scope